8.x/core-concepts/annotations.md

```diff
@@ -2,4 +2,423 @@
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        return 0;
+    }
+}
+```
+
+Change this line into the `TestAnnotations class`:
+
+```php{8}
+class TestAnnotations extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'larammerce:annotation';
+    ...
+}
+```
+
+Insert this code into the `public function handle`:
+
+```php{3}
+public function handle()
+{
+    dd("salam");
+}
+```
+
+Run the following command into the terminal:
+
+```bash{1}
+php artisan larammerce:annotation
+```
+
+**OUTPUT**
+
+![handle-salam](/02.png)
+
+In the path `/larammerce/app/Console/Commands/TestAnnotations.php`, construct a `ReflectiveClass` from the `TestAnnotations class` and get the DocBlock above this class:
+
+
+```php{5,12,13}
+<?php
+
+namespace App\Console\Commands;
+use Illuminate\Console\Command;
+use App\Utils\Reflection\ReflectiveClass;
+
+class TestAnnotations extends Command
+{
+    ...
+    public function handle()
+    {
+        $testAnnotationsRef = new ReflectiveClass(TestAnnotations::class);
+        dd($testAnnotationsRef->getComment());
+    }
+}
+```
+
+**OUTPUT**
+
+![no-docblock](/03.png)
+
+This output indicates that there is no DocBlock above the TestAnnotations class:
+
+```php{1}
+//No DocBlock here.
+class TestAnnotations extends Command
+{
+    ...
+}
+```
+
+Now put this DocBlock above the TestAnnotations class:
+
+```php{1-3}
+/**
+ * @salam(ali="2")
+ */
+class TestAnnotations extends Command
+{
+    ...
+}
+```
+
+**OUTPUT**
+
+s![mere-comment](/04.png)
+
+The output returns a mere comment. In order to get a processed result, change the code in the `public function handle`:
+
+```php{4}
+public function handle()
+{
+    $testAnnotationsRef = new ReflectiveClass(TestAnnotations::class);
+    dd($testAnnotationsRef->getAnnotations());
+}
+```
+
+**OUTPUT**
+
+![processed-comment](/05.png)
+
+As you see, the comment is processed and represented in detail.
+
+Now change the DocBloc above the `TestAnnotations class`:
+
+```php{2}
+/**
+ * @salam(Ali="2", Test=3)
+ */
+class TestAnnotations extends Command
+{
+    ...
+}
+```
+
+**OUTPUT**
+
+![accurate-result](/06.png)
+
+As you see, the result is accurate. So you can write some descriptions above a class and then use them in your code.
+
+Now make a new method called `testFunction` with its related DocBlock into the `TestAnnotations class`:
+
+```php{9-15}
+class TestAnnotations extends Command
+{
+    ...
+        public function __construct()
+    {
+        parent::__construct();
+    }
+    
+    /**
+     * @rules(username="required", password="required|max:20")
+     */
+    public function testFunction()
+    {
+        echo "salam";
+    }
+    ...
+}
+```
+
+Put the codes below in the path `/larammerce/app/Console/Commands/TestAnnotations.php`:
+
+```php{6,14,15}
```

Comment:
The wrong lines have been highlighted.
these are the correct lines:
``` php{6,11,12,13}
...

```

Reply:
line 6 -> use App\Utils\Reflection\ReflectiveMethod;
line 14 -> $testFunctionRef = new ReflectiveMethod(TestAnnotations::class, "testFunction");
line 15 -> dd($testFunctionRef->hasAnnotation("salam"));

These lines should be highlighted and this is true in my system. Please check it again.